pandas/core/series.py

```diff
@@ -1171,6 +1171,71 @@ def _set_value(self, label, value, takeable=False):
         return self
     _set_value.__doc__ = set_value.__doc__
 
+    @Substitution(
+        klass='Series', other_klass='DataFrame',
+        params=dedent("""\
+        arrays : array or list of arrays
+            Either a Series, Index, MultiIndex, list, np.ndarray or a list
+            containing only Series, Index, MultiIndex, list, np.ndarray.\
+        """),
+        examples=dedent("""\
+        >>> s = pd.Series(range(10, 13))
+        >>> s
+        0    10
+        1    11
+        2    12
+        dtype: int64
+
+        Set the index to become `['a', 'b', 'c']`:
+
+        >>> s.set_index(['a', 'b', 'c'])
+        a    10
+        b    11
+        c    12
+        dtype: int64
+
+        Create a MultiIndex by appending to the existing index:
+
+        >>> s.set_index(['a', 'b', 'c'], append=True)
+        0  a    10
+        1  b    11
+        2  c    12
+        dtype: int64
+
+        Create a MultiIndex by passing a list of arrays:
+
+        >>> t = (s ** 2).set_index([['a', 'b', 'c'], ['I', 'II', 'III']])
+        >>> t
+        a  I      100
+        b  II     121
+        c  III    144
+        dtype: int64
+
+        Apply index from another object (of the same length!):
+
+        >>> s.set_index(t.index)
+        a  I      10
+        b  II     11
+        c  III    12
+        dtype: int64
+        """)
+    )
+    @Appender(generic.NDFrame.set_index.__doc__)
+    def set_index(self, arrays, append=False, inplace=False,
+                  verify_integrity=False):
+
+        if not isinstance(arrays, list) or all(is_scalar(x) for x in arrays):
```

Comment:
What case does the first condition handle here? I'm having trouble finding a case where `Series.set_index([foo])` but `Series.set_index(foo)` doesn't (or wouldn't without the `[]`).

Reply:
@TomAugspurger 
The first condition makes sure the second one can execute (i.e. iteration). If it is not true and we end up wrapping it in a list, it will fail in the next step.

The point that is crucial though is that *only* lists are affected by this (and not all list-likes), because `NDFrame.set_index` would try to interpret a list of scalars as a list of keys.

In any case, I added some comments to clarify this behaviour.